src/pages/index.js

```diff
@@ -51,41 +66,140 @@ enableValidation(options);
 //Functions to pass to class objects
 function submitProfile (evt) {
   evt.preventDefault();
-  user.setUserInfo(profileFormPopup.getInputValues());
-  profileFormPopup.close();
+  buttons.profileSubmitButton.textContent = 'Saving...'
```

Comment:
Needs correction: The code for rendering loading gets duplicated.

You can manage loading via `PopupWithForm`:

```js
constructor(popupSelector, { loadingButtonText, ... }) { 
  ...
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
```

```js
// index.js
profileEditPopup.showLoading()
...
.finally(() => {
 profileEditPopup.hideLoading()
})
```